artnet/ksy/artpoll.ksy

```diff
@@ -0,0 +1,115 @@
+  A Controller broadcasts an ArtPoll packet to IP address 2.255.255.255 (sub-net mask 255.0.0.0) at UDP port 0x1936, this is the Directed Broadcast address.
+  The Controller may assume amaximum timeout of 3 seconds between sending ArtPoll and receiving all ArtPollReply packets. If the Controller does not receive a response in this time, it should consider the Node to have disconnected.
+  The Controller that broadcasts an ArtPoll should also reply to its own message (by unicast) with an ArtPollReply. It is a requirement of Art-Net that all controllers broadcast an ArtPoll every 2.5 to 3 seconds. This ensures that any network devices can easily detect a disconnect.
+
+  ## Multiple Controllers
+
+  Art-Net allows and supports multiple controllers on a network. When there are multiple controllers, Nodes will receive ArtPolls from different controllers which may contain conflicting diagnostics requirements. This is resolved as follows:
+  If any controller requests diagnostics, the node will send diagnostics. (ArtPoll->Flags->2).
+  If there are multiple controllersrequesting diagnostics, diagnostics shall be broadcast. (Ignore ArtPoll->Flags->3).
+  The lowest minimum value of Priority shall be used. (Ignore ArtPoll->DiagPriority).
+
+  ## Targeted Mode
+
+  Targeted mode allows the ArtPoll to define a range of Port-Addresses. Nodes will only reply to the ArtPoll is they are subscribed to a Port-Address that is inclusively in the range TargetPortAddressBottom to TargetPortAddressTop. The bit field ArtPoll->Flags->5 is used to enable Targeted Mode.
+
+  ## Minimum Packet Length
+
+  Consumers of ArtPoll shall accept as valid a packet of length 12 octets or larger. Any missing fields are assumed to be zero. This requirement is due to the fact that the length of ArtPoll has increased over the life of the protocol.
+
+  All:
+    Receive: Send ArtPollReply
+    Unicast Transmit: Allowed, with Targeted Mode.
+    Directed Broadcast: Controller broadcasts this packet to poll all Controllers and Nodes on the network
+    Limited Broadcast: Not recommended
+
+seq:
+  - id: id
+    contents: [0x41, 0x72, 0x74, 0x2d, 0x4e, 0x65, 0x74, 0x00]
+    doc: | 
+      Array of 8 characters, the final character is a null termination.
+      Value = 'A' 'r' 't' '-' 'N' 'e' 't' 0x00
+
+  - id: opcode
+    contents: [0x00, 0x20]
+    doc: The OpCode defines the class of data following ArtPoll within this UDP packet. Transmitted little endian. Set to OpPoll
+
+  - id: proto_version
+    contents: [0x00, 0x0e]
+    doc: Protocol version (constant 14, transmitted big endian)
+
+  - id: flags
+    type: flags
+    doc: Set behaviour of Node
+    if: _io.size >= 13
+
+  - id: diag_priority
+    type: u1
+    enum: priority_codes
+    if: _io.size >= 14
```

Comment:
I think we should remove the `if:` if it produces documentation that implies to the reader that this data is optional. If the only impact is a more resilient parser then I'd argue it can remain.

Reply:
Yeah will definitely remove the if on these, given that any packet less than 14 is a violation of the protocol I don't think there should be any special handling around it unless we see actual cases of packets that do this. Honestly I think I got confused